crates/sargon-uniffi/src/system/sargon_os/sargon_os.rs

```diff
@@ -0,0 +1,72 @@
+use std::sync::Once;
+
+use crate::prelude::*;
+use sargon::SargonOS as InternalSargonOS;
+use sargon::Bios as InternalBios;
+
+/// The Sargon "Operating System" is the root "manager" of the Sargon library
+/// which holds an in-memory Profile and a collection of "clients" which are
+/// created from "drivers" which the hosts (iOS/Android wallets) "installs"
+/// during app launch, enabling the  Sargon "Operating System" to e.g read/write
+/// to secure storage and make use of the network connection of the iPhone/Android
+/// phone.
+#[derive(uniffi::Object)]
+pub struct SargonOS {
+    pub(crate) wrapped: Arc<InternalSargonOS>,
+}
+
+#[uniffi::export]
+impl SargonOS {
+    #[uniffi::constructor]
+    pub async fn boot(bios: Arc<Bios>) -> Arc<Self> {
+        let internal_bios: InternalBios = bios.as_ref().clone().into();
+        let internal_sargon_os = InternalSargonOS::boot(Arc::new(
+            internal_bios,
+        ))
+        .await;
+        Arc::new(SargonOS {
+            wrapped: internal_sargon_os,
+        })
+    }
+
+    pub async fn new_wallet(&self) -> Result<()> {
+        self.wrapped.new_wallet().await.into_result()
```

Comment:
@GhenadieVP we might wanna `impl Deref<Target = InternalSargonOS> for SargonOS`, which will allow `self.new_wallet().await.into_result()`, i.e. we can skip `.wrapped` for all methods. which might we worth it. And not only here, potentially in other places inside of `sargon-uniffi`.

Reply:
@GhenadieVP there is also the [`delegate` crate](https://crates.io/crates/delegate) (super cool crate) [which we are using in `Decimal`](https://github.com/radixdlt/sargon/blob/8540902716af79fd375eff3a749b84a384decdcc/crates/sargon/src/core/types/decimal192.rs#L456)

which might allow use to forward ALL methods in a very convenient manner, **with the `into_inner` calls**, something like:

```rust
#[uniffi::export]
impl SargonOS {
    delegate! {
        to self.wrapped {
            pub async fn new_wallet(&self) -> Result<()>;
            pub async fn import_wallet(&self, profile: &Profile,  bdfs_skipped: bool) -> Result<()>;
// just list the signature! :D 
        }
    }
} 
```

Reply:
Sadly this does not work as we have discussed, but I have plans of writing a custom macro that can achieve the same, but in a separate PR :)